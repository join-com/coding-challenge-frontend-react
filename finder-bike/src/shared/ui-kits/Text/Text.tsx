import React from 'react';
import styled from 'styled-components';
import * as Colors from '../Variables/Colors';

export interface IText {
  children?: React.ReactNode;
  onClick?: (e: any) => void;
  color?: string;
  isBold?: boolean;
  href?: string;
  size?: number;
  isBlock?: boolean;
}

const TextWithoutLink = styled.span<IText>`
  color: ${props => Colors[props.color] || Colors.black};
  font-weight: ${props => (props.isBold ? 'bold' : 'normal')};
  font-size: ${({ size }) => (size ? `${size}px` : '1rem')};
  display: ${({ isBlock }) => (isBlock ? 'block' : 'inline-block')};
`;

const TextWithLink = styled.a`
  text-decoration: none;
  ${TextWithoutLink}
`;

function Text(props: IText) {
  if (props.href) return <TextWithLink {...props} />;

  return <TextWithoutLink {...props} />;
}

export default Text;
