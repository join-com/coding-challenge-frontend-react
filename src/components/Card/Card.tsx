import React from "react";
import { Incident } from "../../types.d";

type Props = {
  incident: Incident;
};

const Card: React.FunctionComponent<Props> = ({ incident }) => {
  return (
    <div>
      {incident.media.image_url_thumb && (
        <img src={incident.media.image_url_thumb} />
      )}
      <br />
      title: {incident.title}
      <br />
      description: {incident.description}
      <br />
      occured at: {incident.occurred_at}
      <br />
      updated at: {incident.updated_at}
      <br />
      address: {incident.address}
      <br />
      <br />
    </div>
  );
};

export default Card;
