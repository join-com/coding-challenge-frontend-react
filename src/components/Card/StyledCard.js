import styled from "styled-components";
import { defaultBorder } from "../../helpers/stylesMixins";
import { indents, fontSizes, defaultIndent, colors } from "../../constants/styles";
import media from "../../helpers/mediaQuery";

export const StyledCard = styled.div`
  ${defaultBorder};
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  margin-bottom: ${indents.mobile}rem;
  padding: ${indents.mobile}rem;
  background-color: ${colors.primary};
  ${media.mobile`
    flex-direction: row;
  `} 
  ${media.tablet`
    padding: ${indents.tablet}rem;
    margin-bottom: ${indents.tablet}rem;
  `} 
  ${media.desktop`
    padding: ${indents.desktop}rem;
    margin-bottom: ${indents.desktop}rem;
  `};
`;

export const BikeTitle = styled.h5`
  font-size: ${fontSizes.large}rem;
  margin-bottom: ${defaultIndent}rem;
`;

export const BikeImage = styled.img`
  width: 100%;
  background-color: ${colors.light};
  ${media.mobile`
    width: 50%;
  `}
  ${media.tablet`
    width: 25%;
  `}
   ${media.desktop`
    width: 20%;
  `};
`;
export const BikeInfo = styled.div`
  ${media.mobile`
    padding-left: ${indents.mobile}rem;
  `} 
  ${media.tablet`
    padding-left: ${indents.tablet}rem;
  `} 
  ${media.desktop`
    padding-left: ${indents.desktop}rem;
  `};
`;
