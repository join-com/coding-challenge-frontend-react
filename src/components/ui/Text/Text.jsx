import React from "react";
import PropTypes from "prop-types";

import StyledText from "./StyledText";

const Text = ({ children, size, bold, uppercase }) => (
  <StyledText uppercase={uppercase} size={size} bold={bold}>
    {children}
  </StyledText>
);

Text.propTypes = {
  size: PropTypes.oneOf(["small", "normal", "large", "extraLarge"]),
  bold: PropTypes.bool,
  uppercase: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired
};

Text.defaultProps = {
  size: "normal",
  bold: false,
  uppercase: false
};

export default Text;
