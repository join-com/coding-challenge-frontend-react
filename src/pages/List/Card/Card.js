import React from 'react'
import styled from 'styled-components'
import { NavLink } from 'react-router-dom'
import bycicleIcon from './bycicle.svg'

const colors = {
  gray: 'gray',
}
// import colors from '../../constants/colors';
// import getFormattedDate from '../../utils/getFormattedDate';
// import { getTextWithDots } from './utils';

const getTextWithDots = text => text
const getFormattedDate = secondsTimestamp =>
  new Date(secondsTimestamp * 1000)
    .toLocaleString('en-US', {
      weekday: 'short',
      day: 'numeric',
      month: 'short',
      year: 'numeric',
    })
    .replace(/,/, '')

const Wrapper = styled.div`
  display: flex;
  border: 1px solid ${colors.gray};
  margin: 12px 0;
`

const Content = styled.div`
  padding: 16px 24px;
  width: 100%;
  display: flex;
  flex-direction: column;
`

const Title = styled.div`
  font-weight: 600;
  font-size: 20px;
  line-height: 1.4;
`

const IMAGE_SIZE = 200

const ImageWrapper = styled.div`
  width: ${IMAGE_SIZE}px;
  height: ${IMAGE_SIZE}px;
  border-right: 1px solid ${colors.gray};
  border-bottom: 1px solid ${colors.gray};
  margin-bottom: -1px;
  flex-shrink: 0;
`

const Image = styled.img`
  height: 100%;
`

const Card = ({
  id,
  title,
  description,
  imageUrlThumb,
  address,
  incidentDate,
}) => {
  const linkPath = `/incident/${id}`

  return (
    <Wrapper>
      <NavLink to={linkPath}>
        <ImageWrapper>
          <Image src={imageUrlThumb || bycicleIcon} />
        </ImageWrapper>
      </NavLink>

      <Content>
        <NavLink to={linkPath}>
          <Title>{title}</Title>
        </NavLink>

        <p>{getTextWithDots(description || 'No description', 315)}</p>

        <p>
          {!!incidentDate && getFormattedDate(incidentDate)}
          {!!(incidentDate && address) && ` - `}
          {address}
        </p>
      </Content>
    </Wrapper>
  )
}

export default Card
