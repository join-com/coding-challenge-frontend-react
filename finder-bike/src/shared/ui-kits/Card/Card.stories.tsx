import { storiesOf } from '@storybook/react';
import React from 'react';
import Card from './Card';
import { Text } from '../Text';
import { Icon } from '../Icon';
import { faClock, faMapMarkedAlt } from '@fortawesome/free-solid-svg-icons';

storiesOf('UI-Kit', module).add('Card', () => (
  <Card thumbnailURL="https://seeclickfix.com/files/issue_images/0137/6393/img-image_480745259766912201753441_square.jpg">
    <Text size={15} isBold isBlock>
      Sidewalk - Damage
    </Text>
    <Text isBlock>
      5 plus inches of dangerous sidewalk looming in the shadows. Needs a ADA
      ramp built or raise the slab next to it to match. Cannot believe this has
      gone soo long without being addressed. It says Emeryville but the street
      signs are green (Oakland). Just a matter of time before someone loses some
      teeth going over bicycle handle bars or breaks a limb. At least start with
      painting it orange and putting a cone by it.
    </Text>
    <Text isBlock>
      <Icon icon={faMapMarkedAlt} /> Address: 123 Binh Thanh, HCM
    </Text>
    <Text isBlock>
      <Icon icon={faClock} /> Occurred at: 12/12/2017
    </Text>
    <Text isBlock>
      <Icon icon={faClock} /> Updated at: 12/12/2017
    </Text>
  </Card>
));
