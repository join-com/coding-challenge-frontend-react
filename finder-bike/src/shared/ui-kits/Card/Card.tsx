import React from 'react';
import styled from 'styled-components'
import * as Colors from '../Variables/Colors'
import { Image } from '../Image';
import { Text } from '../Text';

const CardWrapper = styled.div`
  display: grid;
  grid-template-columns: 150px 5fr;
  background-color: ${Colors.white};
  border: 1px solid ${Colors.grey4};
`

const CardContent = styled.div`
  display: grid;
  grid-row-gap: 15px;
  padding: 10px;
`

export interface ICard {
  thumbnailURL: string;
  title: string;
  description: string;
  date: string;
  titleLink?: string;
}

const Card: React.FC<ICard> = (props) =>  {
  return (
    <CardWrapper>
      <Image width={150} height={150} src={props.thumbnailURL} />
      <CardContent>
        <Text size={15} isBold isBlock href={props.titleLink}>{props.title}</Text>
        <Text isBlock>{props.description}</Text>
        <Text isBlock>{props.date}</Text>
      </CardContent>
    </CardWrapper>
  )
}

export default Card;