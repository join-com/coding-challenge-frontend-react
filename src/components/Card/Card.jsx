import React from "react";
import formatTimeStampToDate from "../../helpers/formatDate";
import { StyledCard, BikeImage, BikeInfo } from "./StyledCard";
import Text from "../ui/Text";
import Heading from "../ui/Heading";
import { Link } from "react-router-dom";
import { incidentPropTypes } from "../../constants/propTypes";

function Card({
  item: {
    id,
    title,
    description,
    address,
    occurred_at,
    updated_at,
    media: { image_url_thumb }
  }
}) {
  return (
    <StyledCard key={id}>
      <BikeImage src={image_url_thumb || `/bicycle.svg`} alt={title} />
      <BikeInfo>
        <Link to={`/case/${id}`}>
          <Heading level={3}>{title}</Heading>
        </Link>
        {description && <Text>{description}</Text>}
        <Text>
          Occurred at: {formatTimeStampToDate(occurred_at)} - {address}
        </Text>
        <Text>Updated at: {formatTimeStampToDate(updated_at)}</Text>
      </BikeInfo>
    </StyledCard>
  );
}

Card.propTypes = {
  item: incidentPropTypes.isRequired
};

export default Card;
