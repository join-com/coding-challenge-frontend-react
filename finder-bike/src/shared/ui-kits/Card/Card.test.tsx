import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';
import { Text } from '../Text';

describe('<Card /> spec', () => {
  it('renders the component', () => {
    const component = render(
      <Card thumbnailURL="https://seeclickfix.com/files/issue_images/0137/6393/img-image_480745259766912201753441_square.jpg">
        <Text size={15} isBold isBlock>
          Sidewalk - Damage
        </Text>
        <Text isBlock>
          5 plus inches of dangerous sidewalk looming in the shadows. Needs a
          ADA ramp built or raise the slab next to it to match. Cannot believe
          this has gone soo long without being addressed. It says Emeryville but
          the street signs are green (Oakland). Just a matter of time before
          someone loses some teeth going over bicycle handle bars or breaks a
          limb. At least start with painting it orange and putting a cone by it.
        </Text>
        <Text isBlock>12/12/2017</Text>
        <Text isBlock>12/12/2017</Text>
      </Card>
    );
    expect(component).toMatchSnapshot();
  });
});
