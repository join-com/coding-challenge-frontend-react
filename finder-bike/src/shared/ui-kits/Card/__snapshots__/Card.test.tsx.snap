// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bxivhb fGyFlA"
      >
        <img
          class="sc-bdVaJa bDWFJH"
          height="150"
          src="https://seeclickfix.com/files/issue_images/0137/6393/img-image_480745259766912201753441_square.jpg"
          width="150"
        />
        <div
          class="sc-ifAKCX bgBBpA"
        >
          <a
            class="sc-htpNat fRIlfl"
            href="https://seeclickfix.com"
            size="15"
          >
            Sidewalk - Damage
          </a>
          <span
            class="sc-bwzfXH cEngQN"
          >
            5 plus inches of dangerous sidewalk looming in the shadows. Needs a ADA ramp built or raise the slab next to it to match. Cannot believe this has gone soo long without being addressed. It says Emeryville but the street signs are green (Oakland). Just a matter of time before someone loses some teeth going over bicycle handle bars or breaks a limb. At least start with painting it orange and putting a cone by it.
          </span>
          <span
            class="sc-bwzfXH cEngQN"
          >
            12/12/2017 848 61st St Emeryville, CA, 94608, USA
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bxivhb fGyFlA"
    >
      <img
        class="sc-bdVaJa bDWFJH"
        height="150"
        src="https://seeclickfix.com/files/issue_images/0137/6393/img-image_480745259766912201753441_square.jpg"
        width="150"
      />
      <div
        class="sc-ifAKCX bgBBpA"
      >
        <a
          class="sc-htpNat fRIlfl"
          href="https://seeclickfix.com"
          size="15"
        >
          Sidewalk - Damage
        </a>
        <span
          class="sc-bwzfXH cEngQN"
        >
          5 plus inches of dangerous sidewalk looming in the shadows. Needs a ADA ramp built or raise the slab next to it to match. Cannot believe this has gone soo long without being addressed. It says Emeryville but the street signs are green (Oakland). Just a matter of time before someone loses some teeth going over bicycle handle bars or breaks a limb. At least start with painting it orange and putting a cone by it.
        </span>
        <span
          class="sc-bwzfXH cEngQN"
        >
          12/12/2017 848 61st St Emeryville, CA, 94608, USA
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
